src/meta/processors/job/BalanceJobExecutor.cpp

```diff
@@ -130,1024 +84,81 @@ nebula::cpp2::ErrorCode BalanceJobExecutor::save(const std::string& k, const std
-    return retCode;
-  }
-
-  while (iter->valid()) {
-    auto key = iter->key();
-    PartitionID partId;
-    memcpy(&partId, key.data() + prefix.size(), sizeof(PartitionID));
-    auto partHosts = MetaKeyUtils::parsePartVal(iter->val());
-    for (auto& ph : partHosts) {
-      hostParts[ph].emplace_back(partId);
-    }
-    totalParts++;
-    iter->next();
-  }
-
-  LOG(INFO) << "Host size: " << hostParts.size();
-  auto key = MetaKeyUtils::spaceKey(spaceId);
-  std::string value;
-  retCode = kvstore_->get(kDefaultSpaceId, kDefaultPartId, key, &value);
-  if (retCode != nebula::cpp2::ErrorCode::SUCCEEDED) {
-    LOG(ERROR) << "Access kvstore failed, spaceId " << spaceId
-               << apache::thrift::util::enumNameSafe(retCode);
-    return retCode;
-  }
-
-  auto properties = MetaKeyUtils::parseSpace(value);
-  if (totalParts != properties.get_partition_num()) {
-    LOG(ERROR) << "Partition number not equals " << totalParts << " : "
-               << properties.get_partition_num();
-    return false;
-  }
-
-  int32_t replica = properties.get_replica_factor();
-  LOG(INFO) << "Replica " << replica;
-  if (dependentOnZone && !properties.get_zone_names().empty()) {
-    auto zoneNames = properties.get_zone_names();
-    int32_t zoneSize = zoneNames.size();
-    LOG(INFO) << "Zone Size " << zoneSize;
-    innerBalance_ = (replica == zoneSize);
-
-    auto activeHostsRet = ActiveHostsMan::getActiveHostsWithZones(kvstore_, spaceId);
-    if (!nebula::ok(activeHostsRet)) {
-      return nebula::error(activeHostsRet);
-    }
-
-    std::vector<HostAddr> expand;
-    auto activeHosts = nebula::value(activeHostsRet);
-    std::vector<HostAddr> lostHosts;
-    calDiff(hostParts, activeHosts, expand, lostHosts);
-    // confirmedHostParts is new part allocation map after balance, it would include newlyAdded
-    // and exclude lostHosts
-    HostParts confirmedHostParts(hostParts);
-    for (const auto& h : expand) {
-      LOG(INFO) << "Found new host " << h;
-      confirmedHostParts.emplace(h, std::vector<PartitionID>());
-    }
-    for (const auto& h : lostHosts) {
-      LOG(INFO) << "Lost host " << h;
-      confirmedHostParts.erase(h);
-    }
-
-    auto zonePartsRet = assembleZoneParts(zoneNames, confirmedHostParts);
-    if (zonePartsRet != nebula::cpp2::ErrorCode::SUCCEEDED) {
-      LOG(ERROR) << "Assemble Zone Parts failed";
-      return zonePartsRet;
-    }
-  }
-
-  totalParts *= replica;
-  return true;
-}
-
-nebula::cpp2::ErrorCode BalanceJobExecutor::assembleZoneParts(
-    const std::vector<std::string>& zoneNames, HostParts& hostParts) {
-  // zoneHosts use to record this host belong to zone's hosts
-  std::unordered_map<std::pair<HostAddr, std::string>, std::vector<HostAddr>> zoneHosts;
-  for (const auto& zoneName : zoneNames) {
-    LOG(INFO) << "Zone Name: " << zoneName;
-    auto zoneKey = MetaKeyUtils::zoneKey(zoneName);
+nebula::cpp2::ErrorCode SpaceInfo::getInfo(GraphSpaceID spaceId, kvstore::KVStore* kvstore) {
```

Comment:
I think `load` or `fill` is a better name. Because you fill/load struct itself instead of returning the space info.

Reply:
> I think `load` or `fill` is a better name. Because you fill/load struct itself instead of returning the space info.
> 
> And, could split the code to segments and add some comments.

is this personal preference?

Reply:
Yes, but it's more readable.

Reply:
> personal preference

ok ,loadInfo is better, I'll change, thanks